packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -64,42 +98,37 @@ const toFilterClause = (queryClause: string) => {
   }
 
   const formattedField = isNestedField(field) ? toJsonBKey(field) : field;
-  return [formattedField, value] as [typeof formattedField, typeof value];
-};
-
-const buildCountryCodeFilter = (allowedCountries: string[], value?: string | string[]) => {
-  if (!value) {
-    return allowedCountries;
-  }
-
-  const filteredAllowedCountries = Array.isArray(value)
-    ? allowedCountries.filter(country => value.includes(country))
-    : allowedCountries.filter(country => value === country);
-
-  if (filteredAllowedCountries.length < 1) {
-    throw new Error('No access to any countries due to user permissions and country_code filter');
-  }
 
-  return filteredAllowedCountries;
+  return [formattedField, operator, value] as [
+    typeof formattedField,
+    typeof operator,
+    typeof value,
+  ];
 };
 
-export const extractFilterFromQuery = (allowedCountries: string[], queryFilter?: string) => {
+export const extractFilterFromQuery = (
+  allowedCountries: string[],
+  queryFilter?: string,
+): EntityFilter => {
   if (!queryFilter) {
     return { country_code: allowedCountries };
   }
 
   const filterClauses = queryFilter.split(CLAUSE_DELIMITER).map(toFilterClause);
   const filter: Writable<NotNullValues<EntityFilter>> = {};
 
-  filterClauses.forEach(([field, value]) => {
+  filterClauses.forEach(([field, operator, value]) => {
     const parsedValue = value.includes(MULTIPLE_VALUES_DELIMITER)
       ? value.split(MULTIPLE_VALUES_DELIMITER)
       : value;
 
-    filter[field] = parsedValue;
+    filter[field] = convertValueToAdvancedCriteria(operator, parsedValue);
   });
 
-  filter.country_code = buildCountryCodeFilter(allowedCountries, filter.country_code);
+  // To always force returning only entities in allowed countries, even if there is country_code filter in the query params.
+  filter[QueryConjunctions.AND] = {
+    country_code: allowedCountries,
+  };
```

Comment:
This is a bit of a hack to merge the allowedCountries filter and the optional country_code filter (especially when we want to do `country_code` `!=`) without having to do any custom logic

Reply:
I've checked and I think it still works, but in a coincident way(?), so I tweaked the code in ResponseBuilder.ts to fit with this. Hopefully that's alright